### DISCUSSION TITLE
No completions incoming (PyCharm, VScode)

### DISCUSSION THREAD

**Discussion by norandom** [community/community#7351] [Copilot Conversations]
Hi,

I cannot get any completions from Copilot either via PyCharm or VSCode.
Could it be that it's not ready yet?

1. PyCharm pro remains stuck forever:

> Retrieving Completions

2. VSCode remains stuck forever:

> Synthesizing 0/10 solutions

3. Haven't tested NeoVim yet

* No Proxy or Firewall. To which endpoint would I test a `curl` command?
* PyCharms seems to require the initial auth workflow each time I want to use Copilot
* I have never seen any completion from either IDE so far, speaking of the Copilot plugins

**Versions:**

> PyCharm 2021.2.3 (Professional Edition)
> Build #PY-212.5457.59, built on October 19, 2021
> Licensed to Marius 
> Subscription is active until December 3, 2021.
> Evaluation purpose only.
> Runtime version: 11.0.12+7-b1504.40 x86_64
> VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
> macOS 10.15.7
> GC: G1 Young Generation, G1 Old Generation
> Memory: 1024M
> Cores: 8
> Non-Bundled Plugins: com.github.copilot (1.0.3)

VSCode:

> Version: 1.62.0
> Commit: b3318bc0524af3d74034b8bb8a64df0ccf35549a
> Date: 2021-11-03T15:24:41.297Z
> Electron: 13.5.1
> Chrome: 91.0.4472.164
> Node.js: 14.16.0
> V8: 9.1.269.39-electron.0
> OS: Darwin x64 19.6.0

**Comment by davecheney** [comment]
In VSCode, what output do you see in the View -> Output; GitHub Copilot window?

**Reply by norandom** [reply]
```
> # function to print hello world
> [INFO] [default] [2021-11-05T09:45:27.468Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
> [INFO] [solutions] [2021-11-05T09:45:37.352Z] prompt: #!/usr/bin/env python3
> # Path: PlayGround/foo.py
> 
> # function to print hello world
> [INFO] [default] [2021-11-05T09:45:37.352Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
```

**Comment by jansorg** [comment]
> PyCharms seems to require the initial auth workflow each time I want to use Copilot

This should be fixed with the latest update, 1.0.3. Please let us know if you're still seeing this.

**Comment by norandom** [comment]
I have played around with this further, and it seems that I cannot retrieve the solutions from my usual setup (VMware Fusion guest).
Outside the VMs I have just tested VSCode's plugin and it seems to work.
